website/content/docs/plugins/containerized-plugins.mdx

```diff
@@ -0,0 +1,253 @@
+        ```
+    
+    b. Rootless Docker install:
+        ```bash
+        mkdir -p ~/.config/docker
+        tee ~/.config/docker/daemon.json  <<EOF
+        {
+          "runtimes": {
+            "runsc": {
+              "path": "/usr/local/bin/runsc",
+              "runtimeArgs": [
+                "--host-uds=all"
+                "--ignore-cgroups"
+              ]
+            }
+          }
+        }
+        EOF
+        systemctl --user restart docker
+        ```
+
+### Building plugin container images
+
+To be compatible with running in a container, plugins must be built with at
+least v1.5.0 of [go-plugin](https://github.com/hashicorp/go-plugin). This can
+be pulled in directly via `go get github.com/hashicorp/go-plugin@latest`, or
+through the SDK from at least v0.10.0: `go get github.com/hashicorp/vault/sdk@latest`.
+
+Plugin container images should run a binary the same way it would be expected
+to run as a plain plugin process. For most plugins this simply means running
+the binary as the entrypoint. For example, for the KV plugin:
+
+1. Build the binary
+    ```bash
+    git clone https://github.com/hashicorp/vault-plugin-secrets-kv.git
+    cd vault-plugin-secrets-kv/
+    GOOS=linux go build -o kv cmd/vault-plugin-secrets-kv/main.go
+    ```
+
+1. Create the following `Dockerfile`:
+    ```Dockerfile
+    FROM gcr.io/distroless/static-debian12
+
+    COPY kv /bin/kv
+
+    ENTRYPOINT [ "/bin/kv" ]
+    ```
+
+1. Build the image:
+    ```bash
+    docker build -t hashicorp/vault-plugin-secrets-kv .
+    ```
+
+The image entrypoint, command, and environment can be further configured at
+registration time by using the `-command`, `-args`, and `-env` flags for
+`vault plugin register`.
+
+### Registration
+
+Once Vault has a suitable container runtime available, the remaining steps are
+very similar to registering and using other external plugins. The config setting
+for [`plugin_directory`](/vault/docs/configuration#plugin_directory) is not required
+for containerized plugins, but any plugin containers must already be available
+locally when mounted.
+
+1. Store the SHA256 of the plugin image:
+    ```bash
+    SHA256="$(docker images --no-trunc --format="{{ .ID }}" hashicorp/vault-plugin-secrets-kv | cut -d: -f2)"
+    ```
+
+1. Register a plugin, specifying your container image as `-oci_image`:
+    ```bash
+    vault plugin register \
+        -sha256="$SHA256" \
+        -oci_image=hashicorp/vault-plugin-secrets-kv \
+      secret kv-container
+    ```
+
+1. Enable the plugin:
+    ```bash
+    vault secrets enable kv-container
+    ```
+
+1. You can now interact with it like any other plugin:
+    ```shell-session
+    $ vault kv put -mount=kv-container hello subject=containers
+    Success! Data written to: kv-container/hello
+
+    $ vault kv get -mount=kv-container hello
+    ===== Data =====
+    Key        Value
+    ---        -----
+    subject    containers
+    ```
+
+### Container lifecycle and metadata
```

Comment:
Everything between here and the Podman instructions should be a separate overview document since we're discussing how the plugins work rather than providing explicit instructions for how to do something.

Reply:
Oh I didn't read this before my comment above. WDYT about where the two documents belong? I think where this document currently is is a good place for reference documentation. Would you put the how-to guide alongside it or somewhere else in the tree entirely?